api/v1beta1/disruption_types.go

```diff
@@ -314,6 +333,33 @@ func (s *DisruptionSpec) validateGlobalDisruptionScope() (retErr error) {
 		retErr = multierror.Append(retErr, errors.New("disk pressure disruptions apply to all containers, specifying certain containers does not isolate the disruption"))
 	}
 
+	// Rule: DisruptionTrigger
+	if s.Triggers != nil {
+		if s.Triggers.Inject != nil && s.Triggers.CreatePods != nil {
+			if !s.Triggers.Inject.NotBefore.IsZero() && !s.Triggers.CreatePods.NotBefore.IsZero() && s.Triggers.Inject.NotBefore.Before(&s.Triggers.CreatePods.NotBefore) {
+				retErr = multierror.Append(retErr, fmt.Errorf("spec.trigger.inject.notBefore is %s, which is before your spec.trigger.createPods.notBefore of %s. inject.notBefore must come after createPods.notBefore if both are specified", s.Triggers.Inject.NotBefore, s.Triggers.CreatePods.NotBefore))
+			}
+		}
+
+		triggerTypes := []*DisruptionTrigger{s.Triggers.Inject, s.Triggers.CreatePods}
+
+		for _, subTrigger := range triggerTypes {
+			if subTrigger == nil {
+				continue
+			}
+
+			if !subTrigger.NotBefore.IsZero() {
+				now := metav1.Now()
+				if subTrigger.NotBefore.Before(&now) {
```

Comment:
How close to now can we be?
How far from now do we allow?

Maybe let's enforce at least 1m from now, and no more than Xh ?

WDYT?

Reply:
We call validateGlobalDisruptionScope on Updates as well as Creates, unfortunately, so we can't compare against time.Now, and we don't have access to the creationTimestamp here. We could move this check to the reconcile loop, but I think maybe we just leave it, and plug the gap if it actually becomes an issue?

Reply:
No problem with this approach 🤗